src/Middleware/RequestThrottling/test/MiddlewareTests.cs

```diff
@@ -82,5 +82,20 @@ public void InvalidArgumentIfMaxConcurrentRequestsIsNull()
             });
             Assert.Equal("options", ex.ParamName);
         }
+
+        [Fact]
+        public async void RequestsBlockedIfQueueFull()
```

Comment:
I think we need to add a few more tests here. Specifically, one that has multiple requests and verifying the status code of 503.

Reply:
Yeah. While not strictly necessary for testing this, it's probably also a good idea to start TestServer-based tests.